devicetypes/keen-home/keen-home-smart-vent.src/keen-home-smart-vent.groovy

```diff
@@ -477,47 +193,24 @@ def configure() {
-            //commenting out switch cluster bind as using wrapper onOffConfig of zigbee class
-        //"zdo bind 0x${device.deviceNetworkId} 1 1 0x0006 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0008 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0402 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0403 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0001 {${device.zigbeeId}} {}", "delay 500"
-
-        // configure report commands
-        // zcl global send-me-a-report [cluster] [attr] [type] [min-interval] [max-interval] [min-change]
-
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // vent on/off state - type: boolean, change: 1
-        // "zcl global send-me-a-report 6 0 0x10 5 60 {01}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
-
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // vent level - type: int8u, change: 1
-        // "zcl global send-me-a-report 8 0 0x20 5 60 {01}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+    def cmds = [
+            zigbee.temperatureConfig(30, 300) +
+            zigbee.onOffConfig() +
+            zigbee.addBinding(zigbee.LEVEL_CONTROL_CLUSTER) +
+            zigbee.addBinding(PRESSURE_MEASUREMENT_CLUSTER) +
+            zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021, DataType.UINT8, 600, 21600, 0x01) // battery precentage
+    ]
 
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // temperature - type: int16s, change: 0xA = 10 = 0.1C
-        // "zcl global send-me-a-report 0x0402 0 0x29 60 60 {0A00}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+    return delayBetween(cmds) + refresh()
+}
 
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // keen home custom pressure (tenths of Pascals) - type: int32u, change: 1 = 0.1Pa
-        // "zcl mfg-code 0x115B", "delay 200",
-        // "zcl global send-me-a-report 0x0403 0x20 0x22 60 60 {010000}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
 
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // battery - type: int8u, change: 1
-        // "zcl global send-me-a-report 1 0x21 0x20 60 3600 {01}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
-    ]
+private boolean isObstructed() {
+    def currentState = device.currentValue("switch")
 
-    return configCmds + zigbee.onOffConfig() + refresh()
+    if (currentState == "obstructed") {
+        sendEvent(name: "level", value: 0) // convert legacy implementation
```

Comment:
I think we should also set `switch` to a valid value here

Reply:
The `switch` and `level` events should also not be displayed

Reply:
I think this was just an error on my part. This was always meant to be a switch event. The old DTH currently sets "switch" to "obstructed" and this is just meant to give it capability-aligning values. It never sets "level" to anything weird.